File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque

def bfs():
  global visited, q, count
  while q:
    cur = q.popleft()
    for next in graph[cur]:
      if not visited[next]:
        visited[next] = True
        q.append(next)
        count+=1

V = int(sys.stdin.readline())
E = int(sys.stdin.readline())
start = 1
count = 0 # 몇 대의 컴퓨터가 감염됐는지
q = deque() # 사용할 큐
graph = [[] for _ in range(V+1)]
visited = [False] * (V+1)

# 입력 받기
for i in range(E):
  a, b = map(int, sys.stdin.readline().strip().split())
  
  # 양방향 입력이니깐
  graph[a].append(b)
  graph[b].append(a)

visited[1] = True
q.append(start)
bfs()
print(count)

File: 백준/Silver/2606. 바이러스/바이러스.java

import java.io.*;
import java.util.*;

public class Main {

	static int N, M;
	static List<Integer>[] computers;
	static boolean[] visited;

	public static void main(String[] args) throws IOException {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

		N = Integer.parseInt(bf.readLine());
		M = Integer.parseInt(bf.readLine());

		computers = new List[N + 1];
		for (int i = 0; i <= N; i++) {
			computers[i] = new ArrayList<>();
		}
		visited = new boolean[N + 1];

		for (int i = 0; i < M; i++) {
			String[] sa = bf.readLine().split(" ");
			int start = Integer.parseInt(sa[0]);
			int end = Integer.parseInt(sa[1]);

			computers[start].add(end);
			computers[end].add(start);
		}

		int result = bfs(1);

		System.out.println(result);

	}

	private static int bfs(int x) {
		int result = 0;
		Queue<Integer> q = new LinkedList<>();
		q.add(x);
		visited[x] = true;

		while(!q.isEmpty()){
			x = q.poll();
			for(int i=0; i<computers[x].size(); i++){
				int next = computers[x].get(i);
				if(visited[next]) continue;
				visited[next] = true;
				q.add(next);
				result++;
			}
		}

		return result;
	}
}